added printer setup function

Pikokosan opened the pull request:
added a printer button as well as the ablility to add a printer via the setup.py

Commit:
Various bugfixes + hardware simulator

Commit:
moved logo

Commit:
Implemented sendmail + bugfixes

laurentalacoque committed:
bugfixes

laurentalacoque committed:
removed utf8 that caused issues + update() status during lengthy operations

laurentalacoque committed:
added update() for lengthy operations

laurentalacoque committed:
fixed upload error because archive moved the file

laurentalacoque committed:
won't fail if GPIO not present

Commit:
added has_buttons and buttons_number methods

Commit:
Create software buttons when no hardware buttons are present

laurentalacoque committed:
Added fullscreen launch

laurentalacoque committed:
fixed typo

laurentalacoque committed:
updated with a delay to be able to launch at startup

laurentalacoque committed:
fixed sendmail fails because image was renamed

laurentalacoque committed:
update gitignore

Commit:
Added ressources

Commit:
Merge branch 'master' of https://github.com/laurentalacoque/TouchSelfie

Added ressources

Commit:
Added comment for my own comprehension

laurentalacoque committed:
a new keyboard inspired from tkkb.py

laurentalacoque committed:
changed to mykb

laurentalacoque committed:
autohide config button

laurentalacoque committed:
update

laurentalacoque committed:
Remove unused file

laurentalacoque committed:
Adapted countdown size to the preview resolution

laurentalacoque committed:
Added install instructions

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

Commit:
make fullscreen optional

laurentalacoque committed:
removed old interface

laurentalacoque committed:
removed old keyboard

laurentalacoque committed:
removed unused test

laurentalacoque committed:
Added command line options for fullscreen, upload and sendemail options

laurentalacoque committed:
Changed keyboard to mykb

laurentalacoque committed:
added credentials as kwargs as well

laurentalacoque committed:
Added a setup program to help configure and arrange Google credentials

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

laurentalacoque committed:
documentation

laurentalacoque committed:
Merge branch 'master' of https://github.com/laurentalacoque/TouchSelfie

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

laurentalacoque committed:
added gif animation example

laurentalacoque committed:
added gif animation example

laurentalacoque committed:
cleaning. Added Hardware button configuration

laurentalacoque committed:
Added command line options to disable config button and hardware button handling

laurentalacoque committed:
updated doc

laurentalacoque committed:
removed unused constants

laurentalacoque committed:
moved constants to constants.py

laurentalacoque committed:
a new assistant for credentials configuration and album selection

laurentalacoque committed:
removed graphical configuration, everything is now in constants.py

laurentalacoque committed:
removed configuration from the main program

laurentalacoque committed:
updated doc

laurentalacoque committed:
removed graphical configuration

laurentalacoque committed:
updated gitignore

laurentalacoque committed:
renamed file

laurentalacoque committed:
updated readme

laurentalacoque committed:
Update README.md

laurentalacoque committed:
first working Google Photos + Sendmail example

laurentalacoque committed:
Now a class with interface simplification

laurentalacoque committed:
added authorization callback example + debug

laurentalacoque committed:
New json-based configuration file

laurentalacoque committed:
replaced legacy authentication with the new oauth2 + implemented configuration

laurentalacoque committed:
removed account email from constructor

laurentalacoque committed:
This becomes useless with full oauth2 authentication

laurentalacoque committed:
useless since oauth2services.py implementation

laurentalacoque committed:
should implement this in oauth2services.py

laurentalacoque committed:
useless since oauth2services.py

laurentalacoque committed:
setup.sh script now creates a launcher script

laurentalacoque committed:
unused

laurentalacoque committed:
Now implemented in configuration.py

laurentalacoque committed:
Revert "useless since oauth2services.py implementation"

This reverts commit 6037f8db9b8acba439bac6c6d3d664a290202e00.

laurentalacoque committed:
Revert "This becomes useless with full oauth2 authentication"

This reverts commit c7ded49da0010f749e71ca736a8b7e13939dd658.

laurentalacoque committed:
implemented in oauth2services.py

laurentalacoque committed:
now implemented in configuration.py

laurentalacoque committed:
setup.sh creates this

laurentalacoque committed:
This is unused

laurentalacoque committed:
removed unused code

laurentalacoque committed:
now exits if not configured

laurentalacoque committed:
updated gitignore

laurentalacoque committed:
added configuration files in a single place

laurentalacoque committed:
Now exit in case of bad configuration

laurentalacoque committed:
New setup program

laurentalacoque committed:
renamed

laurentalacoque committed:
adapted to renaming

laurentalacoque committed:
Added info so that git creates the directory

laurentalacoque committed:
moved constructor parameters to config

laurentalacoque committed:
Added connection tests

laurentalacoque committed:
typo

laurentalacoque committed:
No test uploads in right album

laurentalacoque committed:
fixed bug in tests

laurentalacoque committed:
typo in send_message

laurentalacoque committed:
Merge branch 'master' into passwordless

laurentalacoque committed:
Merge pull request #1 from laurentalacoque/passwordless

Passwordless authentication for emails and photo uploads

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Added long-press configuration

laurentalacoque committed:
updated .gitignore

laurentalacoque committed:
Merge pull request #2 from laurentalacoque/runtime_config

New quicksettings window

laurentalacoque committed:
Now uses full_screen property of configuration

laurentalacoque committed:
bugfix for hardware buttons

laurentalacoque committed:
documentation

laurentalacoque committed:
Update constants.py

laurentalacoque committed:
Update fakehardware.py

laurentalacoque committed:
Documentation

laurentalacoque committed:
Documentation

laurentalacoque committed:
Documentation + bugfix in hardware_buttons

laurentalacoque committed:
Merge pull request #21 from laurentalacoque/documentation

Documentation

laurentalacoque committed:
Added page-based graphical setup

laurentalacoque committed:
colouring

laurentalacoque committed:
coded options dependent screen changing

laurentalacoque committed:
Icons for graphical setup

laurentalacoque committed:
Authentication process is stable

laurentalacoque committed:
added (useless) google_photo_album_name property (nicer in configuration stages

laurentalacoque committed:
updated

laurentalacoque committed:
#5 First version of working graphical install

laurentalacoque committed:
improved UI for album selection

laurentalacoque committed:
resolved #19 Optionnal soft keyboard for configuration

laurentalacoque committed:
Merge pull request #22 from laurentalacoque/graphical-install

Graphical install

laurentalacoque committed:
Catch Tk error and defaults gracefully to console, console-based now saves photoalbum

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Auto-hide and display again cursor based on mouse motion

laurentalacoque committed:
Auto-hide and display again cursor based on mouse motion

laurentalacoque committed:
Merge branch 'pointer-and-keyboard' of https://github.com/laurentalacoque/TouchSelfie into pointer-and-keyboard

laurentalacoque committed:
Added keyboard shortcuts to event. Configurable in constants.py

laurentalacoque committed:
Merge pull request #24 from laurentalacoque/pointer-and-keyboard

Pointer and keyboard

laurentalacoque committed:
Solved #25 => now creates photobooth.sh startup script

Improved startup script so that it can propagate arguments
run.log is now photobooth.log in the main directory

laurentalacoque committed:
Merge pull request #27 from laurentalacoque/graphical-install

Solved #25 => now creates photobooth.sh startup script

laurentalacoque committed:
Added constant image list for picture countdown

laurentalacoque committed:
Prepare for add_overlay in fake hardware

laurentalacoque committed:
example countdown pictures to use as preview overlay

laurentalacoque committed:
First try at overlayed countdown to enable horizontal flip

laurentalacoque committed:
Bugfixes, mirrored preview is functionnal

laurentalacoque committed:
Merge pull request #28 from laurentalacoque/mirrored-preview

Mirrored preview is functionnal

laurentalacoque committed:
Configuration and setup ready for mail logging

laurentalacoque committed:
sendmail.log file configurable in constants.py

laurentalacoque committed:
Added a consent screen and log_email code

laurentalacoque committed:
Hopefully fixes longpress event triggered incorrectly

laurentalacoque committed:
merged master's bug resolution

laurentalacoque committed:
Bugfixes

laurentalacoque committed:
Merge pull request #31 from laurentalacoque/email-logging

Email logging with user consent screen
Also fixes bug #32 (for sure now)

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Fixed centering of overlay

laurentalacoque committed:
Merge branch 'master' of https://github.com/laurentalacoque/TouchSelfie

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Quickfix for Authentication refresh doesn't work for email sending

laurentalacoque committed:
Merge branch 'master' of https://github.com/laurentalacoque/TouchSelfie

laurentalacoque committed:
Quickfix for pictures title incomplete in Google Photos albums

laurentalacoque committed:
Fixed gmtime instead of Localtime

laurentalacoque committed:
[Quickfix #38] Incorrect default value for GPIO-less systems

Incorrect default value for systems without GPIO when used outside of user_interface.py

laurentalacoque committed:
Quickfix for bug #38

Missed this incorrect instanciation of hardware_buttons on GPIOless systems

laurentalacoque committed:
Countdown image size now depends of a constant

laurentalacoque committed:
overlay size now based on screen size (hopefully solves #37)

laurentalacoque committed:
[Overlay size and position] Fixed #37

BugFix #37 : adapt overlay image size to the preview

laurentalacoque committed:
Info on email from protected networks

laurentalacoque committed:
#41 Layout adapts to screen size

laurentalacoque committed:
Merge pull request #42 from laurentalacoque/big-screens-layout-#41

#41 Layout adapts to screen size

laurentalacoque committed:
Solves #40: now uses copy/delete whenever os.rename fails (e.g. usb disks)

laurentalacoque committed:
Merge pull request #43 from laurentalacoque/#40-cross-fs-move

Solves #40: now uses copy/delete whenever os.rename fails (e.g. usb disks)

Pikokosan committed:
Printer Function added

added a printer function with some configureation options. TODO: added printer selection to the setup.py. currently defaults to printer #2 as #1 is VNC on my test system

Pikokosan committed:
added print enable/disable to setup.py

No printer selection added but does have the option to disable on page 1

Pikokosan committed:
printer setup added 

added printer adding to the setup.py and made it a configureable item

Pikokosan committed:
printer setup and failsafes

checks for cups installation when running the setup.py as well as redraws page #0 when printer enable is selected.

Pikokosan committed:
forgot to save

Pikokosan committed:
Self not removed from user_interface.py

Pikokosan committed:
Added exception for CUPS

This is a test

Pikokosan committed:
Redraw page

This should redraw page 0 and remove the enable printer option so users can finish there setup if they forgot to setup CUPS

Pikokosan committed:
message box fixed

Pikokosan committed:
Merge pull request #1 from Pikokosan/print_dev

printer exceptions and failsafe

laurentalacoque committed:
Merge pull request #46 from Pikokosan/master

This merge adds a button to print snapshots on a preconfigured printer using cups/python-cups

laurentalacoque committed:
Hotfix after printer feature added

Fixed pycups -> python-cups package name in README
Fixed enable_printer was saved to true even when no printer configured (cups runtime error)

laurentalacoque committed:
Added Print feature

laurentalacoque committed:
Support for printer shortcut

laurentalacoque committed:
Improved documentation

laurentalacoque committed:
added 'enable_effects' configuration option

laurentalacoque committed:
Added effects button icon

laurentalacoque committed:
Added effects thumbnails

laurentalacoque committed:
Added IMAGE_EFFECTS dict of parameters

laurentalacoque committed:
Implemented effect selection and processing

laurentalacoque committed:
Updated documentation (image effects)

laurentalacoque committed:
hotfix: image effect wasn't reset between shots

laurentalacoque committed:
auto-centering of thumbnails in effect selection window

laurentalacoque committed:
fixed centering bug for last row/col

laurentalacoque committed:
Changed effects order, added some effects and params

laurentalacoque committed:
tool to help generate effect thumbnails

laurentalacoque committed:
bugfix

laurentalacoque committed:
new effects thumbs

laurentalacoque committed:
enlarged thumbs

laurentalacoque committed:
bugfix

laurentalacoque committed:
added short documentation for generate_effects_thumbnails.py

laurentalacoque committed:
Added image effect configuration option

laurentalacoque committed:
Image Effects

Support of PiCamera builtin image effects
This definitely closes #44

laurentalacoque committed:
changed panel image

laurentalacoque committed:
Update README.md

laurentalacoque committed:
updated for thumbnail generation and sendmail.log

laurentalacoque committed:
Make it more clear that 'none' is no effect

laurentalacoque commented:
Hi @wyojustin, @osbock 

Thanks for your nice piece of code! 👍 

I developped a few major changes with a valuable contribution by @Pikokosan 
We wish to share this with you for possible insertion into the original TouchSelfie

Thanks!

Here is a list of the changes:

### Zero password stored
- Now integrally based on OAuth2, neither the send-email, nor the upload-pictures will ask for and store a password
- Using OAuth2 also allows people to send mails from netwoks that block smtp (e.g. schools/work)

### Easier setup
- a new `setup.sh` assistant will guide you through the configuration of the features you need (send_email, auto-upload) and will help you install the Google credentials.
- credentials installation now points to a google-hosted wizard to simplify the application-id creation

### Hardware and software buttons support for snapshot effects (None/Four/Animation)
- Added GPIO hardware interface for three buttons (with connections in hardware/ directory). Each button triggers a different effect. Software buttons are added if GPIO is not available

### Print selfies immediately
- A new 'print' button allows you to print selfies on a preconfigured printer (courtesy [Pikokosan](https://github.com/Pikokosan))

### New effects
- Added "Animation" effect that produces animated gifs (needs imagemagick)


- Choose among `PiCamera` builtin image effects for more creative snapshots


### Higher resolution pictures
- supports new v2.1 pi camera resolutions
- Supports arbitrary resolution for snapshots (configure it in constants.py)

### Better snap preview mode

- Preview now uses images for countdown: be creative! see `constants.py` for customizations.
- Preview is now horizontally flipped which is consistent with cameraphone behaviors and your bathroom mirror. (Previous mode was confusing for users)

### User interface improvements

- Modern "black" userinterface with icon buttons
- Interface is screen-size independant
- New skinnable Touchscreen keyboard "mykb.py"

- Removed configuration button to avoid pranks
  - use `setup.sh` to setup credentials and setup Google Photo Album
  - tune configuration in `constants.py`
  - limited runtime configuration (exit, enable/disable email and upload) is still possible via long press)

- Mail address logging available in configuration
  - users can opt-out from the mail logging if activated

jephilosoraptor committed:
Revised to include a QWERTY layout

jephilosoraptor committed:
Delete mykb.py

jephilosoraptor committed:
Add a QWERTY layout in mykb.py

laurentalacoque committed:
Querty layout for keyboard

Add QWERTY Layout to mykb.py

laurentalacoque committed:
Update README.md

Added oauth2client update following a [comment](https://github.com/hs-kn/TouchSelfie-extended/commit/4fedcbf75a19da5e06da50e2961227e296f1fdb7#commitcomment-31248586)

Thanks to @hs-kn

laurentalacoque committed:
inserted_items counting to fix bad filtering

This hopefully fixes #55

laurentalacoque committed:
Merge pull request #57 from laurentalacoque/bad-filtering-of-albums-#55

inserted_items counting to fix bad filtering

laurentalacoque committed:
keyboard shortcuts now inactive when snap is active, this solves #59

laurentalacoque committed:
Isolated upload and save in try/except blocks to solve #53

laurentalacoque committed:
Improve #53 by checking if a picture was taken but not uploaded nor saved

laurentalacoque committed:
Update README.md

Commit:
make software buttons inactive during processing to solve #59

laurentalacoque committed:
Declaration of intent

Commit:
Work in progress, now can list albums and send emails. MISSING: photo upload

Commit:
File upload now functionnal

Commit:
changed client_id and credentials name to avoid confusion. updated setup

Commit:
fixed: old name for client secret file

Commit:
Added album creation. Changed scope to .appendonly

Commit:
added selection and creation of photos albums

Commit:
Failproof upload: if transfer fails because of bad album id, simply upload to users's library

Commit:
fixed: album_id=None wouldn't work

Commit:
Setup can now <create> and <select> photo albums

Commit:
updated documentation

Commit:
updated documentation

laurentalacoque committed:
Implemented the new 'Google Photos Library API'

Move from Picasa Web API to Photoslibrary API: fixes #60

Commit:
Added snapshots for the wiki page

laurentalacoque committed:
Improved doc

Commit:
fixed invalid_scope error

Commit:
Merge branch 'master' of https://github.com/laurentalacoque/TouchSelfie

Commit:
moved to logging

Commit:
catch and throw authentication errors

Commit:
improved logging

Commit:
improved logging

Commit:
Improved logging

laurentalacoque committed:
Implemented standard logging, fixes #64

Change debug messages from print to more standard logging module

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Update security and privacy infos

laurentalacoque committed:
Update README.md

laurentalacoque committed:
Added Caroline's video

laurentalacoque committed:
implemented usb key detection and saving

laurentalacoque committed:
logging from the start, added fake Color class to cv2_camera

laurentalacoque committed:
test for directory existence

laurentalacoque committed:
bugfix

laurentalacoque committed:
added configuration flag to disable/enable usb key archiving

laurentalacoque committed:
Write snaphots to every plugged-in USB drives (fixes #68)

Write snapshots to every writeable USB drives plugged into the raspberry fixes #68

laurentalacoque committed:
setup.sh +x

laurentalacoque committed:
Won't fail if logo.png absent

laurentalacoque committed:
logging from the start / fixed gif dep

wyojustin commented:
Awesome work moving from Picasa to Google photos!

laurentalacoque commented:
Thanks go to @carolinedunn, she asked me to 😉
Big thank you for the original contribution!

wyojustin commented:
@laurentalacoque I've made a couple more updates you might be interested in.

I tried to make a pull request, but failed. Thanks for your pull request, I've accepted all your changes and have made some updates myself. This "issue" is just a way to say thanks and offer the changes back.

Justin

Changes:
added email validation so we don't attempt to send to mal-formed email addresses
mykb: added cancel key (send turns to cancel when email is not valid)
mykb: added keyboard shortcuts for escape and send
added Nine collage feature.
added software button "order" attribute so you can arrange buttons on bottom of screen.

laurentalacoque commented:
Thanks Justin, I'll take a look at these 😉

carolinedunn commented:
Thank you for featuring my video!

wyojustin commented:
Team effort!!! Super video!

Justin

On Mon, Feb 18, 2019 at 10:12 PM Caroline Dunn <notifications@github.com>
wrote:

> Thank you for featuring my video!
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/wyolum/TouchSelfie/pull/19#issuecomment-464963330>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA3QW39B1Bl0IE36gTGv8TiJ3_SFjL9Aks5vO2u0gaJpZM4V0zNp>
> .
>


-- 
Justin
